avm/res/databricks/workspace/main.bicep

```diff
@@ -156,14 +170,26 @@ resource cMKManagedDiskKeyVault 'Microsoft.KeyVault/vaults@2023-07-01' existing
   }
 }
 
-resource workspace 'Microsoft.Databricks/workspaces@2023-02-01' = {
+// This var is a workaround for the fact both properties can not be null or empty, they either exist or not in the properties object.
+var defaultStorageFirewallProperties = !empty(privateStorageAccount)
```

Comment:
Don, I am not comfortable with this change here. Please can you rather find a way **not** to introduce this new "var" as its used only once. Can you rather try to include it in the properties just as managedResourceGroupId was done? Give it some thought and see if you can make it work

Reply:
Hey @donheerschap,
also had a brief look. Might make sense to embedd the variable's code in the same way the inner properties are build. Pushes the complexity there, but one doesn't have to scroll around searching where to code is coming from 😄 A matter of personal taste, I guess 😉

Reply:
This is how it should work:
![image](https://github.com/user-attachments/assets/8097cedd-a57d-4ff3-a13f-658ea0119fb4)

Reply:
@clintgrove It cannot be done as the way managedResourceGroupId is done. If the property exists but is null or an empty object, which is in the case for everyone who doesn't want (or didn't have in the past) deploy this setting, the deployment will fail. The union is the only way it prevents from breaking in both cases. 

@AlexanderSehr has the same implementation except the definition of the var is just embedded in the union, which is indeed a matter of taste, but to keep consistent with the properties.parameters objects, it's the preferred to change to.

Reply:
I think that's what @clintgrove was aiming at too (:

It really is a shame that there is still no 'undefined' value or similar available in ARM/Bicep/Azure that would allow us to avoid these `union()` statements alltogether. 🤔